homeassistant/components/home_connect/light.py

```diff
@@ -115,88 +109,107 @@ class HomeConnectLight(HomeConnectEntity, LightEntity):
-        self._custom_color_key: str | None
-        self._color_key: str | None
-        if ambient:
-            self._brightness_key = BSH_AMBIENT_LIGHT_BRIGHTNESS
-            self._custom_color_key = BSH_AMBIENT_LIGHT_CUSTOM_COLOR
-            self._color_key = BSH_AMBIENT_LIGHT_COLOR
-            self._attr_color_mode = ColorMode.HS
-            self._attr_supported_color_modes = {ColorMode.HS}
-        else:
-            self._brightness_key = COOKING_LIGHTING_BRIGHTNESS
-            self._custom_color_key = None
-            self._color_key = None
-            self._attr_color_mode = ColorMode.BRIGHTNESS
-            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+        def get_setting_key_if_setting_exists(setting_key: str | None) -> str | None:
+            if setting_key and setting_key in device.appliance.status:
+                return setting_key
+            return None
+
+        self._brightness_key = get_setting_key_if_setting_exists(desc.brightness_key)
+        self._custom_color_key = get_setting_key_if_setting_exists(
+            desc.custom_color_key
+        )
+        self._color_key = get_setting_key_if_setting_exists(desc.color_key)
+        self._enable_custom_color_value_key = desc.enable_custom_color_value_key
+        self._custom_color_key = get_setting_key_if_setting_exists(
+            desc.custom_color_key
+        )
+        self._brightness_scale = desc.brightness_scale
+
+        match (self._brightness_key, self._custom_color_key):
+            case (None, None):
+                self._attr_color_mode = ColorMode.ONOFF
+                self._attr_supported_color_modes = {ColorMode.ONOFF}
+            case (_, None):
+                self._attr_color_mode = ColorMode.BRIGHTNESS
+                self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+            case (_, _):
+                self._attr_color_mode = ColorMode.HS
+                self._attr_supported_color_modes = {ColorMode.HS, ColorMode.RGB}
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Switch the light on, change brightness, change color."""
-        if self._ambient:
-            _LOGGER.debug("Switching ambient light on for: %s", self.name)
-            try:
-                await self.hass.async_add_executor_job(
-                    self.device.appliance.set_setting, self.bsh_key, True
-                )
-            except HomeConnectError as err:
-                _LOGGER.error("Error while trying to turn on ambient light: %s", err)
-                return
-            if ATTR_BRIGHTNESS in kwargs or ATTR_HS_COLOR in kwargs:
+        _LOGGER.debug("Switching light on for: %s", self.name)
+        try:
+            await self.hass.async_add_executor_job(
+                self.device.appliance.set_setting, self.bsh_key, True
+            )
+        except HomeConnectError as err:
+            _LOGGER.error("Error while trying to turn on light: %s", err)
+            return
```

Comment:
For a followup PR (this is probably unused in home_connect, so it might be bigger), if its an issue that the user made, raise `ServiceValidationError`, otherwise `HomeAssistantError`. Those are shown to the user so they at least get some feedback on why this failed